### PR TITLE
Ptrdist/anagram: Checked Strings

### PR DESCRIPTION
**Pull request by lenary**
This depends on https://github.com/Microsoft/checkedc/pull/239

The good news is the only _Unchecked parts left are:
- printf, fprintf
- setjmp, longjmp

Inserted some dynamic bounds cast to cast from an array_ptr to a nt_array_ptr. This could be wrong, please make sure it's the right approach. The bounds in this file are all whacky, we should be using nt_array_ptr in far more places and i will revisit it after I've finished making everything else checked.

### CODE REVIEW

**Comment by lenary** [comment]
@dtarditi this is waiting on a review.

**Comment by lenary** [comment]
This is ready for review again, now includes unchecked blocks.

**Comment by lenary** [comment]
Oh this may not build immediately, as hacks.h was in the other branch. Once merged it will.

File: MultiSource/Benchmarks/Ptrdist/anagram/anagram.c
Comment (dtarditi) [inline MultiSource/Benchmarks/Ptrdist/anagram/anagram.c]: This cast should be wrapped in an unchecked block: casts from _Array_ptr to _Nt_array_ptr won't be allowed.   The compiler work to add this checking is tracked by the existing work item https://github.com/Microsoft/checkedc-clang/issues/391.

**Review by dtarditi — APPROVED** [review]
One suggested change.  Otherwise - looks good.

File: MultiSource/Benchmarks/Ptrdist/anagram/anagram.c
Comment (lenary) [inline MultiSource/Benchmarks/Ptrdist/anagram/anagram.c:513]: Then this won't be either

Comment (lenary) [reply MultiSource/Benchmarks/Ptrdist/anagram/anagram.c:513]: Oh it has an unchecked block because we couldn't prove the bounds anyway.

Comment (dtarditi) [reply MultiSource/Benchmarks/Ptrdist/anagram/anagram.c:513]: Yes, that's why I didn't mention it.

**Pull request merged**
